compiler/ir/ir.psi2ir/src/org/jetbrains/kotlin/psi2ir/generators/ReflectionReferencesGenerator.kt

```diff
@@ -84,27 +82,27 @@ class ReflectionReferencesGenerator(statementGenerator: StatementGenerator) : St
     }
 
     fun generateCallableReference(
-        startOffset: Int,
-        endOffset: Int,
+        ktElement: KtElement,
         type: KotlinType,
         callableDescriptor: CallableDescriptor,
         typeArguments: Map<TypeParameterDescriptor, KotlinType>?,
         origin: IrStatementOrigin? = null
-    ): IrCallableReference =
-        when (callableDescriptor) {
-            is FunctionDescriptor ->
-                generateFunctionReference(
-                    startOffset, endOffset, type,
-                    context.symbolTable.referenceFunction(callableDescriptor.original),
-                    callableDescriptor,
-                    typeArguments,
-                    origin
-                )
-            is PropertyDescriptor ->
-                generatePropertyReference(startOffset, endOffset, type, callableDescriptor, typeArguments, origin)
+    ): IrCallableReference {
+        val startOffset = ktElement.startOffsetSkippingComments
+        val endOffset = ktElement.endOffset
+        return when (callableDescriptor) {
+            is FunctionDescriptor -> {
+                val symbol = context.symbolTable.referenceFunction(callableDescriptor.original)
+                generateFunctionReference(startOffset, endOffset, type, symbol, callableDescriptor, typeArguments, origin)
+            }
+            is PropertyDescriptor -> {
+                val mutable = get(BindingContext.VARIABLE, ktElement)?.isVar ?: true
```

Comment:
I suppose we could use `getOrFail` here

Reply:
The frontend doesn't actually generate fake variables for references used in delegated properties' accessors, so `getOrFail` fails on those.

Reply:
Oh, you're right